sycl/test/check_device_code/matrix/matrix-nvptx-bf16-test.cpp

```diff
@@ -0,0 +1,279 @@
+// REQUIRES: gpu, cuda
+
+// RUN: %clangxx -fsycl-device-only -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --cuda-gpu-arch=sm_80 -DSYCL_EXT_ONEAPI_MATRIX=3 -S -Xclang -emit-llvm %s -o -| FileCheck %s
+
+#include <CL/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+
+constexpr int stride = 16;
+
+int main() {
+
+  buffer<uint16_t, 1> bufA(nullptr, range<1>(1));
+  buffer<uint16_t, 1> bufB(nullptr, range<1>(1));
+  buffer<float, 1> bufC(nullptr, range<1>(1));
+  buffer<float, 1> bufD(nullptr, range<1>(1));
+
+  queue q;
+
+  q.submit([&](handler &cgh) {
+    auto accC = bufC.get_access<access::mode::read_write>(cgh);
+    auto accA = bufA.get_access<access::mode::read_write>(cgh);
+    auto accB = bufB.get_access<access::mode::read_write>(cgh);
+    auto accD = bufD.get_access<access::mode::read_write>(cgh);
+
+    cgh.parallel_for<class row_row_m16n16k16>(
+        nd_range<2>({1, 32}, {1, 32}), [=
```

Comment:
what is the size of the original buffer? 
what does 32 in the second dimension of the global range represents?

Reply:
This is just a device code test that doesn't execute so I don't worry about the size of the buffer, I only care that it has the correct type.  It is for this reason that I tried to make it as simple as possible: the purpose is to check that the runtime is calling the correct builtin for each case.  Also check out the integration tests here: https://github.com/intel/llvm-test-suite/pull/650

32 is used in the second dimension of the global range because I only use one subgroup: the meaning of 32 in the second dimension of the global range carries the same meaning as the second dimension of the local range: the number of nd_items per subgroup that is required for the matrix operations for the cuda backend.